### PR TITLE
Add xcm

### PR DESCRIPTION
**Pull request by N1ghtStorm**
Here added:
- XCM Config and XCM executor config
- Added orml xTokens pallet for assets sending via XCM
- Added converter
- Added transactor

### CODE REVIEW

File: common/Cargo.toml [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+scale-info = { version = "2.1.1", default-features = false, features = ["derive"] }
+frame-support = { git = "https://github.com/paritytech/substrate", default-features = false, branch = "polkadot-v0.9.28" }
+
+[features]
+default = ["std"]
+std = [
+	"codec/std",
+	"scale-info/std",
+	"frame-support/std",
+]
```
Comment (wer1st) [inline common/Cargo.toml]: add empty line

File: pallets/converter/README.md [hunk 2]
```diff
@@ -0,0 +1 @@
+
```
Comment (wer1st) [inline pallets/converter/README.md]: Why is it empty?

File: pallets/converter/src/weights.rs [hunk 3]
```diff
@@ -0,0 +1,85 @@
...
+	}
+
+	fn change_multilocation_mapping() -> Weight {
+		EXTRINSIC_FIXED_WEIGHT
+	}
+
+	fn delete_mapping() -> Weight {
+		EXTRINSIC_FIXED_WEIGHT
+	}
+}
```
Comment (wer1st) [inline pallets/converter/src/weights.rs]: Add empty line

File: pallets/transactor/README.md [hunk 4]
```diff
@@ -0,0 +1 @@
+
```
Comment (wer1st) [inline pallets/transactor/README.md]: Also empty

File: pallets/xcm-app/src/benchmarking.rs [hunk 5]
```diff
@@ -0,0 +1,146 @@
...
+// INCLUDING, BUT NOT LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR
+// A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL Polka Biome Ltd. BE LIABLE FOR ANY
+// DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING,
+// BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS;
+// OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT,
+// STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE
+// USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+
+use super::*;
+#[allow(unused)]
```
Comment (wer1st) [inline pallets/xcm-app/src/benchmarking.rs]: Probably we don't need it anymore

File: pallets/xcm-app/src/impls.rs [hunk 6]
```diff
@@ -0,0 +1,165 @@
...
+	fn ensure_can_withdraw(
+		_currency_id: Self::CurrencyId,
+		_who: &T::AccountId,
+		_amount: Self::Balance,
+	) -> sp_runtime::DispatchResult {
+		log::trace!(
+			target: "xcm::XCMApp",
+			"ensure_can_withdraw",
+		);
+		fail!(Error::<T>::MethodNotAvailible)
```
Comment (wer1st) [inline pallets/xcm-app/src/impls.rs]: Probably we need TODO comments in all places we want to implement later. Just to not lost something

Comment (N1ghtStorm) [reply pallets/xcm-app/src/impls.rs]: some of these functions could not ever be implemented

File: pallets/xcm-app/src/lib.rs [hunk 7]
```diff
@@ -0,0 +1,441 @@
...
+					MultilocationToAssetId::<T>::remove(multilocation.clone());
+					Self::deposit_event(Event::<T>::MappingDeleted(asset_id, multilocation));
+				},
+			};
+			Ok(().into())
+		}
+
+		#[pallet::weight(<T as Config>::WeightInfo::delete_mapping())]
+		#[frame_support::transactional]
+		pub fn fake_transfer(
```
Comment (vovac12) [inline pallets/xcm-app/src/lib.rs:386]: I think we can remove or make it root extrinsic

**Review by vovac12 — APPROVED** [review]

**Review by wer1st — APPROVED** [review]

**Pull request merged**